[DSP] Key Performance Indicators Dashboard w/ Sigma (#25)
hydradatabase/docs: guides/key-performance-indicators-with-sigma.md

  

  

---

  

description: Sigma is a cloud-native analytics platform that uses a familiar spreadsheet interface to give business users instant access to explore and get insights from their cloud data warehouse. It requires no code or special training to explore billions of rows, augment with new data, or perform “what if” analysis on all data in real⁠-⁠time.

  

---

  

  

> # Integrating Hydra with [Sigma](https://www.sigmacomputing.com/)

Reviewer: ```suggestion
# Integrating Hydra with Sigma
```

Please move this link elsewhere rather than in the title, I believe it won't come through in the title. I think you can put markdown links in the `description` metadata.

  
## Scope

The focus of this guide is to showcase the ease of integrating Hydra with Sigma by developing several KPI metrics based on data within a Hydra database.

To cater to a large audience, the data and KPI's were chosen to be as agnostic as possible. The data used is mock sales data from IBM, and the metrics made are the following:
* YoY Revenue Growth (card)
* YoY Net Profit Growth (card)
* QoQ Revenue Growth (time-series line chart)

Your KPI's will most likely differ, but this will hopefully serve as a foundation for the development of more metrics.
  

## Setup

  

  

- Ensure you have the details of your Hydra instance handy.

  

- Sign up for a free trial at [Sigma](https://www.sigmacomputing.com/free-trial).

  

  

### Building the Table with Hydra

  

1. We will use sample sales data from [IBM](https://relational.fit.cvut.cz/dataset/GOSales). Head to the following [repository](https://github.com/IBM/watson-machine-learning-samples) and download the .csv located in `watson-machine-learning-samples/cpd4.6/data/go_sales/go_daily_sales.csv`.

  

> 2. Connect to your Hydra database and upload the .csv. If you're unfamiliar with the process of connecting via. psql and creating/populating a table, please follow this [guide](https://docs.hydra.so/centralize-data/load/from-local-csv-file).

Reviewer: Can you expand the instructions for loading the CSV? I think this is small enough that it's worth putting the instructions inline rather than referencing other documentation.

Author: Sure. Found here: 
https://github.com/AmirZahre/hydra_docs/edit/main/guides/key-performance-indicators-with-sigma.md#building-the-table-with-hydra

2. Connect to your Hydra database and upload the .csv. We will be following the steps outlined in the following [guide](https://docs.hydra.so/centralize-data/load/from-local-csv-file).

  
	a. Grab the `psql` command from the Hydra dashboard and get into the `psql` console.
	
	b. Due to the differences in date mapping between the `.csv` and the `DATE` data-type when creating the table, we will temporarily set the **datestyle** to `euro`.

	```sql
	set datestyle = euro;
	show datestyle;
	 DateStyle 
	-----------
	 ISO, DMY
	```
	c. The below code is used to create the sales_data table. Note: we will create a [columnar](https://docs.hydra.so/organize/data-modeling/row-vs-column-tables) table, which makes it much quicker to obtain all data for a particular column.

	```
	CREATE TABLE sales_data (

	retailer_code INT,

	product_number INT,

	order_method_code INT,

	date DATE,

	quantity INT,

	unit_price FLOAT,

	unit_sale_price FLOAT

	) USING columnar;
	```

	  d. Once the table has been created, upload the .csv by utilizing the following command:
	  
	  `\COPY sales_data FROM 'notebooks/data/ibm_gosales/go_daily_sales.csv' DELIMITER ',' CSV HEADER;`

  

  

3. You should now have a sales-oriented table populated with approximately 150,000 rows.

    * The following code is used to create the sales_data table:

      ```
      CREATE TABLE sales_data (
          retailer_code INT,
          product_number INT,
          order_method_code INT,
          date TIMESTAMP,
          quantity INT,
          unit_price FLOAT,
          unit_sale_price FLOAT

>           )

Reviewer: This table is relatively small at 150k rows, but should the user be using columnar for the table?

Author: sure! quick fix.

      ```

  

3. You should now have a sales-oriented table populated with approximately 150,000 rows.

  

  

### Connecting Sigma with Hydra

  

  

Now we can configure Sigma to access our sample data in Hydra.

  

  

1. As mentioned earlier, create a [trial account](https://www.sigmacomputing.com/free-trial).

  

2. Open your Admin Portal by selecting **Administration** in the user menu at the top right of your screen.

  

3. Select the **Connections** page from the top left-hand panel.

  

4. Click the **Create Connection** button on the top right.

  

5. Choose a name for the connection (we went with **hydra-sales_data_2023**) and select PostgreSQL as the type of database you'd like to connect to.
    

>     ![Step_1](https://user-images.githubusercontent.com/71795488/227726542-f797223a-ba13-46ab-8640-507101f2fa3f.png)

Reviewer: I love the screenshots, but would you mind moving the screenshots to inside the repo? You can create a subfolder like `.sigma-images` for this purpose. [Here's an example of using images from within the repo.](https://github.com/hydradatabase/docs/blob/804de88712bbec9ceb956d82ad1251192519ca9b/organize/data-hygiene/dbt-transformation-and-model-management.md?plain=1#L5)

Author: I created a sub-folder within `/assets` to hold the screenshots: `/hydra_docs/.gitbook/assets/.sigma-images`. Train of thought of creating a sub-folder is to keep them organized and separate from the other images as there are **a lot** of screenshots. Let me know if you'd just like them dumped in `/assets/` instead.

> 

Reviewer: I'm not sure why but there's a very large amount of unneeded newlines in this file. You do not need more than two newlines between any section.  A regexp search and replace will fix these:

Search: `\n\n\n+` 
Replace: `\n\n`

There are a few cases where extra newlines will still need to removed manually.

Author: This was a weird one, and the only explanation I could think of was the spacing being added when copying/pasting the content of the .md to/from the markdown editor I regularly use (stackedit). Easy fix, got rid of them all.

---

  

  

  

description: Sigma is a cloud-native analytics platform that uses a familiar spreadsheet interface to give business users instant access to explore and get insights from their cloud data warehouse. It requires no code or special training to explore billions of rows, augment with new data, or perform “what if” analysis on all data in real⁠-⁠time.

  

  

  

---

  

  

  

  

# Integrating Hydra with Sigma

  

  

## Scope

  

  

The focus of this guide is to showcase the ease of integrating Hydra with [Sigma](https://www.sigmacomputing.com/) by developing several KPI metrics based on data within a Hydra database.

  

  

> To cater to a large audience, the data and KPI's were chosen to be as agnostic as possible. The data used is mock sales data from IBM, and the metrics made are the following:

Reviewer: ```suggestion
To cater to a large audience, the data and KPIs were chosen to be as agnostic as possible. The data used is mock sales data from IBM, and the metrics are the following:
```

  

* YoY Revenue Growth (card)

  

* YoY Net Profit Growth (card)

  

* QoQ Revenue Growth (time-series line chart)

  

  

> Your KPI's will most likely differ, but this will hopefully serve as a foundation for the development of more metrics.

Reviewer: ```suggestion
Your KPIs will likely differ, but this will hopefully serve as a foundation for the development of more metrics.
```

  

  

## Setup

  

  

  

  

- Ensure you have the details of your Hydra instance handy.

  

  

  

- Sign up for a free trial at [Sigma](https://www.sigmacomputing.com/free-trial).

  

  

  

  

### Building the Table with Hydra

  

  

  

> 1. We will use sample sales data from [IBM](https://relational.fit.cvut.cz/dataset/GOSales). Head to the following [repository](https://github.com/IBM/watson-machine-learning-samples) and download the .csv located in `watson-machine-learning-samples/cpd4.6/data/go_sales/go_daily_sales.csv`.

Reviewer: ```suggestion
1. We will use sample sales data from [IBM](https://relational.fit.cvut.cz/dataset/GOSales). [Head to the repository](https://github.com/IBM/watson-machine-learning-samples/tree/master/cpd4.6/data/go_sales) and download the `go_daily_sales.csv` file from the linked directory (`cpd4.6/data/go_sales`).
```

> 2. Connect to your Hydra database and upload the .csv. We will be following the steps outlined in the following [guide](https://docs.hydra.so/centralize-data/load/from-local-csv-file).

Reviewer: ```suggestion
2. Connect to your Hydra database and import the CSV file. We will be following the steps for importing a CSV outlined in the [documentation](https://docs.hydra.so/centralize-data/load/from-local-csv-file).
```

  
	a. Grab the `psql` command from the Hydra dashboard and get into the `psql` console.
	b. Due to the differences in date mapping between the `.csv` and the `DATE` data-type when creating the table, we will temporarily set the **datestyle** to `euro`.

	```sql
	set datestyle = euro;
	show datestyle;
	 DateStyle 
	-----------
	 ISO, DMY
	```

> 	c. The below code is used to create the sales_data table. Note: we will create a [columnar](https://docs.hydra.so/organize/data-modeling/row-vs-column-tables) table, which makes it much quicker to obtain all data for a particular column.

Reviewer: ```suggestion
	c. The below code is used to create the sales_data table. Note: we will create a [columnar table](https://docs.hydra.so/organize/data-modeling/row-vs-column-tables), which makes it much quicker to obtain all data for a particular column.
```

	```sql
	CREATE TABLE sales_data (

	retailer_code INT,

	product_number INT,

	order_method_code INT,

	date DATE,

	quantity INT,

	unit_price FLOAT,

	unit_sale_price FLOAT

> 	) USING columnar;

Reviewer: ```suggestion
	CREATE TABLE sales_data (
	retailer_code INT,
	product_number INT,
	order_method_code INT,
	date DATE,
	quantity INT,
	unit_price FLOAT,
	unit_sale_price FLOAT
	) USING columnar;
```

	```

	  d. Once the table has been created, upload the .csv by utilizing the following command:

> 	  `\COPY sales_data FROM 'notebooks/data/ibm_gosales/go_daily_sales.csv' DELIMITER ',' CSV HEADER;`

Reviewer: ```suggestion
	  `\COPY sales_data FROM 'go_daily_sales.csv' DELIMITER ',' CSV HEADER;`
```

  

  

3. You should now have a sales-oriented table populated with approximately 150,000 rows.

  

  

  

  

### Connecting Sigma with Hydra

  

  

  

  

Now we can configure Sigma to access our sample data in Hydra.

  

  

  

  

> 1. As mentioned earlier, create a [trial account](https://www.sigmacomputing.com/free-trial).

Reviewer: ```suggestion
1. Start by creating a [trial account](https://www.sigmacomputing.com/free-trial), or logging in if you have already done so.
```

  

  

  

5. Choose a name for the connection (we went with **hydra-sales_data_2023**) and select PostgreSQL as the type of database you'd like to connect to.

  

![](/.gitbook/assets/.sigma-images/connection_details.png)

  
  

  

  

  

6. Retrieve your database credentials from your Hydra dashboard. You will need Hostname, User, Password, and Database. Keep all other options unchanged. Once all the information has been entered, select **Create** on the top right.

  

![](/.gitbook/assets/.sigma-images/connection-credentials.png)

  

  

  

  

7. Once created, you will be able to view the connection details.

  

![](/.gitbook/assets/.sigma-images/connection-summary.png)

  
  
  

  

### Creating Your First Hydra-backend Dashboard

  

  

  

1. Back on the main page, click **Create New** on the top left followed by selecting **Workbook**. You will see a banner entitled **DATA ELEMENTS** on the toolbar to the left.

  

  

  

2. Within the aforementioned toolbar, select **VIZ** followed by clicking on **TABLES AND DATASETS**. This will bring you to the Hydra database and will show you the available schemas. Your data should exist within the **public** schema.

  

> ![](/.gitbook/assets/.sigma-images/select-table.png)

Reviewer: ```suggestion
  ![](/.gitbook/assets/.sigma-images/select-table.png)
```

I think indenting the images will put the images inside each `li` element of the ordered list here (applies to each image in this section and the previous section) which should make the steps a little easier to follow.

Author: Done. All images referencing an `li` element should now be inside their respective element.

  

  

  

  

3. Navigate to the aforementioned schema, select **sales_data**, and then confirm your selection by clicking the button on the bottom right.

  

  

  

  

### Congratulations! You've connected Hydra with Sigma! What next?

  

  

> Now that your data is connected, you're able to utilize your Hydra tables and create insights using Sigma's visualization. Let's experiment with creating the below dashboard showcasing some important KPI's relating to YoY revenue and profit.

Reviewer: ```suggestion
Now that your data is connected, you're able to utilize your Hydra tables and create insights using Sigma's visualization. Let's experiment with creating the below dashboard showcasing some important KPIs relating to YoY revenue and profit.
```
